### PR TITLE
Refine SSO configure prompt titles

### PR DESCRIPTION
**Pull request by AndrewAsseily**
*Issue #, if available:*
* CLI-3756

*Description of changes:*
* Remove "CLI" from the prompts that apply to all SDKs
* Clarify that the default output format is json, not just "None"
* Changed the example message to address the following issue: https://github.com/aws/aws-cli/issues/9132
* Included unit tests for the aforementioned changes in addition to manual testing

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: awscli/customizations/configure/sso.py [hunk 1]
```diff
@@ -488,6 +488,10 @@ def _prompt_for_cli_output_format(self):
 
     def _prompt_for_profile_config(self, config_name, text, completions=None):
         current_value = self._profile_config.get(config_name)
+
+        if config_name == 'output' and not current_value:
+            current_value = 'json'
```
Comment (ashovlin) [inline awscli/customizations/configure/sso.py:493]: Is this going to persist `output = json` in the profile which we don't do today?

If so, I'd try to avoid a change in behavior with a quick cosmetic change, largely so we don't have to hunt down all of the implications. 
1. If using `current_value` -> `new_value`, is there a way to not persist it if the user just pressed enter to accept the default.
2. Or, can we just communicate the default behavior through the `text` and `prompt_fmt` strings, then still actually collect and "save" `None`?

**Review by ashovlin — CHANGES_REQUESTED** [review]

File: awscli/customizations/configure/sso.py
Comment (AndrewAsseily) [inline awscli/customizations/configure/sso.py:1]: Alternative to updating the prompt text:
 
In `_prompt_for_profile_config`, 

```
    # Only save if user actually typed something different
    if new_value and new_value != current_value:
        self._new_profile_config_values[config_name] = new_value
 ```

File: tests/unit/customizations/configure/test_sso.py [hunk 2]
```diff
@@ -1513,6 +1514,37 @@ def test_single_account_single_role_device_code_fallback(
...
+        assert f"Successfully configured SSO for profile: {profile_name}" in captured.out
+
+    def test_print_conclusion_default_profile_case_insensitive(selfself, sso_cmd, capsys):
+        sso_cmd._print_conclusion(True, 'DEFAULT')
+        captured = capsys.readouterr()
+        assert "The AWS CLI is now configured to use the default profile." in captured.out
+        assert "aws sts get-caller-identity" in captured.out
+
+    def test_print_conclusion_empty_profile_name(self, sso_cmd, capsys):
+        sso_cmd._print_conclusion(True, '')
```
Comment (ashovlin) [inline tests/unit/customizations/configure/test_sso.py:1544]: Is this even realistic? Doesn't empty fallback to default?

**Review by ashovlin — APPROVED** [review]

File: tests/unit/customizations/configure/test_sso.py (hunk 2 above)
Comment (AndrewAsseily) [reply tests/unit/customizations/configure/test_sso.py:1544]: After manual testing, I confirmed that an empty profile does indeed fall back to the default profile. I'll remove this test since it's not testing a valid edge case.
